app/importers/file_importers/star_math_importer.rb

```diff
@@ -63,16 +63,18 @@ def import_row(row)
       return
     end
 
-    star_assessment = StudentAssessment.where({
+    star_assessment = StudentAssessment.find_or_initialize_by(
       student_id: student.id,
       date_taken: date_taken,
       assessment: star_mathematics_assessment
-    }).first_or_create!
+    )
 
-    star_assessment.update!({
+    star_assessment.assign_attributes(
```

Comment:
This method matches the `assign_attributes` used in `mcas_row.rb`.

Reply:
FWIW, what motivates the factoring in `McasRow` is that it only does one read.  And then the importer uses `RecordSyncer` to do the write, and to track which Insights records are updated, created or left unchanged.

Reply:
Gotcha! That would be a good pattern to follow for the STAR importers.